includes/Commerce/Orders.php

```diff
@@ -82,10 +82,86 @@ public function create_local_order( Order $remote_order ) {
 	 * @param Order $remote_order Orders API order object
 	 * @param \WC_Order $local_order local order object
 	 * @return \WC_Order
+	 * @throws SV_WC_Plugin_Exception|\WC_Data_Exception
 	 */
 	public function update_local_order( Order $remote_order, \WC_Order $local_order ) {
 
-		// TODO: implement
+		// add/update items
+		foreach ( $remote_order->get_items() as $item ) {
+
+			$wc_product_id = $item['retailer_id'];
+
+			$matching_wc_order_item = false;
+
+			// check if the local order already has this item
+			foreach ( $local_order->get_items() as $wc_order_item ) {
+
+				if ( $wc_order_item instanceof \WC_Order_Item_Product && $wc_product_id === (string) $wc_order_item->get_product_id() ) {
+					$matching_wc_order_item = $wc_order_item;
+					break;
+				}
+			}
+
+			if ( empty( $matching_wc_order_item ) ) {
+
+				$wc_product = wc_get_product( $wc_product_id );
+
+				if ( ! $wc_product instanceof \WC_Product ) {
+					throw new SV_WC_Plugin_Exception( "Product with WC ID $wc_product_id not found" );
+				}
+
+				$matching_wc_order_item = new \WC_Order_Item_Product();
+				$local_order->add_item( $matching_wc_order_item );
+			}
+
+			$matching_wc_order_item->set_product_id( $wc_product_id );
+			$matching_wc_order_item->set_quantity( $item['quantity'] );
+			$matching_wc_order_item->set_subtotal( $item['quantity'] * $item['price_per_unit']['amount'] );
+			// TODO: should we use the estimated_tax or the captured_tax on the line below?
+			$matching_wc_order_item->set_subtotal_tax( $item['tax_details']['estimated_tax']['amount'] );
+			$matching_wc_order_item->save();
+		}
+
+		// update information from selected_shipping_option
+		$selected_shipping_option = $remote_order->get_selected_shipping_option();
+
+		$local_order->set_shipping_total( $selected_shipping_option['price']['amount'] );
+		$local_order->set_shipping_tax( $selected_shipping_option['calculated_tax']['amount'] );
+
+		// update information from shipping_address
+		$shipping_address = $remote_order->get_shipping_address();
+
+		$local_order->set_shipping_last_name( $shipping_address['name'] );
```

Comment:
Why just the order's last name here?

Reply:
Because we only have one `name` field, so we have nothing to set in the first name.

Reply:
Right but in the [doc examples](https://developers.facebook.com/docs/commerce-platform/order-management/order-api) for buyer name (and very likely shipping name), they show a full name `John Smith`. I think we could at least split the string after the first space to try and separate out first and last. That won't cover all cases, but it should for a lot, and then next steps would be:
1. Test against a real API response and confirm that assumption, that the full name is returned
1. Bring the issue to Facebook to see if they can return distinct first and last

Reply:
From their reply on Slack:

> 2. I don't believe there is a guarantee it will be the customer's full name. IIRC for IG it can sometimes by the IG user's handle. I also just confirmed on FB I am able to enter any arbitrary text in the Name field for the Delivery address.

I believe we don't have a lot of options here. Do you think we should split after the first space if there is a space?

Reply:
yeah I think that's the only thing we can do really, just try and find the first space

Reply:
Updated in https://github.com/facebookincubator/facebook-for-woocommerce/pull/1515/commits/a495851b6b6e52c5bf03e612c207c8981d595907.